lib/matplotlib/afm.py

```diff
@@ -196,21 +226,22 @@ def _parse_char_metrics(fh):
         # Split the metric line into a dictionary, keyed by metric identifiers
         vals = dict(s.strip().split(' ', 1) for s in line.split(';') if s)
         # There may be other metrics present, but only these are needed
-        if not {'C', 'WX', 'N', 'B'}.issubset(vals):
+        if not required_keys.issubset(vals):
             raise RuntimeError('Bad char metrics line: %s' % line)
         num = _to_int(vals['C'])
         wx = _to_float(vals['WX'])
         name = vals['N']
         bbox = _to_list_of_floats(vals['B'])
         bbox = list(map(int, bbox))
+        metrics = CharMetrics(wx, name, bbox)
         # Workaround: If the character name is 'Euro', give it the
         # corresponding character code, according to WinAnsiEncoding (see PDF
         # Reference).
         if name == 'Euro':
             num = 128
         if num != -1:
-            ascii_d[num] = (wx, name, bbox)
-        name_d[name] = (wx, bbox)
+            ascii_d[num] = metrics
+        name_d[name] = metrics
```

Comment:
I guess this is technically a backcompat break? (someone could have been doing `wx, bbox = name_d[name]`.)
It would not have been a problem if `parse_afm` was private (... as it probably should have been) but a note is probably useful.

Reply:
If we are at it, shouldn't we just make `parse_afm` private? After all, `parse_afm` is not something I would want to maintain as part of the public matplotlib API in the future.

Reply:
I'm fine with deprecating it.